ts/create-kpt-functions/src/cmd/type_create.ts

```diff
@@ -87,6 +67,76 @@ export async function typeCreate(packageDir: string) {
   log(format.finishMarker(desc));
 }
 
+// chooseTypeSource lets the user choose the source for the kubernetes types and returns
+// numeric choice
+function chooseTypeSource(): number {
+  const defaultTypeSource = 0;
+  const ch = validator.getValidString(
+    () =>
+      question(
+        `> What is the source of OpenAPI document defining the Kubernetes types? This will be used to generate Typescript library (${defaultTypeSource})
+[0] Use built-in document (No cluster required)
+[1] Download the document from a Kubernetes cluster
+Please enter your numeric choice: `
+      ),
+    validator.isEmptyOrMaxInt(1),
+    defaultTypeSource.toString()
+  );
+  log(`Using Type source choice "${ch}".\n`);
+
+  return Number(ch);
+}
+
+// getOptsForTypeSource gets the url request options based on the numeric choice input
+// refer to chooseTypeSource() for choice outcomes
+function getOptsForTypeSource(typeSource: number): request.Options {
+  let url = ``;
+  let opts: request.Options = {
+    url,
+  };
+  switch (typeSource) {
+    case 0:
+      log('Using built-in kubernetes types');
+      url = BUILTIN_OPENAPI_URL;
+      opts = {
+        url,
+      };
+      return opts;
+
+    case 1:
+      // Get the kubeconfig context the user wants to use.
+      const kc = new KubeConfig();
+      kc.loadFromDefault();
+      const contexts = kc.contexts;
+      if (contexts.length == 0) {
+        log(
+          failure(
+            'No contexts found in kubeconfig file. Please set a context entry in kubeconfig.'
+          )
+        );
+        process.exit(1);
+      }
+      const currentContext = kc.currentContext;
+      const contextIdx = chooseContext(contexts, currentContext);
+      const useContext = contexts[contextIdx];
+      const cluster = kc.clusters.find(c => c.name === useContext.cluster);
+      if (!cluster) {
+        throw new Error('Cluster for specified context not found.');
+      }
+      kc.setCurrentContext(useContext.name);
+      // set the url to cluster openAPI if cluster exists
+      url = `${cluster.server}/openapi/v2`;
+      opts = {
+        url,
+      };
+      kc.applyToRequest(opts);
+      return opts;
+
+    default:
+      throw new Error('Invalid choice for Kubernetes types source.');
```

Comment:
Wordsmithing so the user knows how to rerun: Invalid choice for Kubernetes types source. Invoke `npm run kpt:type-create` to source Kubernetes types.